src/inmanta/data/__init__.py

```diff
@@ -1571,6 +1571,26 @@ async def insert(self, connection: Optional[asyncpg.connection.Connection] = Non
         )
         await self._execute_query(query, *values, connection=connection)
 
+    async def insert_with_overwrite(self, connection: Optional[asyncpg.connection.Connection] = None) -> None:
+        """
+        Insert a new document based on the instance passed. If the document already exists, overwrite it.
+        """
+        (column_names, values) = self._get_column_names_and_values()
+        column_names_as_sql_string = ",".join(column_names)
+
+        values_as_parameterize_sql_string = ",".join(["$" + str(i) for i in range(1, len(values) + 1)])
+        primary_key_fields = self._get_names_of_primary_key_fields()
+        primary_key_string = ",".join(primary_key_fields)
+        update_set = list(set(column_names) - set(self._get_names_of_primary_key_fields()))
+        update_set_string = ",\n".join([f"{item} = EXCLUDED.{item}" for item in update_set])
+        query = f"""INSERT INTO {self.table_name()}
+            ({column_names_as_sql_string})
+            VALUES ({values_as_parameterize_sql_string})
+            ON CONFLICT ({primary_key_string})
+            DO UPDATE SET
+            {update_set_string};"""
```

Comment:
This is nitpicking, but I find this can be made even more readable: (same for the other query)
```suggestion
        query = f"""
            INSERT INTO {self.table_name()}
            ({column_names_as_sql_string})
            VALUES ({values_as_parameterize_sql_string})
            ON CONFLICT ({primary_key_string})
            DO UPDATE SET
            {update_set_string};
        """
```